Sources/Utils/Rng/MobileCoinSeedableRng.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  Copyright (c) 2020-2021 MobileCoin. All rights reserved.
+//
+
+// swiftlint:disable unused_setter_value
+import Foundation
+
+public class MobileCoinSeedableRng: MobileCoinRng {
+    let _seed: Data32
+
+    var seed: Data32 {
+        _seed
+    }
+
+    init(seed: Data32) {
+        self._seed = seed
+    }
+
+    override convenience init() {
+        let bytesCount = 32
+        var randomBytes = [UInt8](repeating: 0, count: bytesCount)
+
+        let status = SecRandomCopyBytes(kSecRandomDefault, bytesCount, &randomBytes)
+        guard status == errSecSuccess else {
+            var message = ""
+            if #available(iOS 11.3, *), let errorMessage = SecCopyErrorMessageString(status, nil) {
+                message = ", message: \(errorMessage)"
+            }
+
+            logger.fatalError("Failed to generate bytes. SecError: \(status)" + message)
+        }
+
+        guard let seedData32 = Data32(Data(randomBytes)) else {
+            // will not happen
+            logger.fatalError("Failed to generate Data32 for rng seed.")
+        }
+
+        self.init(seed: seedData32)
+    }
+
+    var wordPos: Data16 {
```

Comment:
if this isnt available on the default implementation, lets remove it. Internally its not needed so it doesn't to be part of the base-class. SDK consumers that need this from the ChaCha20 implementation can still get it because the reference they hold onto will be typed.

Reply:
Removed MobileCoinSeedableRng abstraction as it doesn't really seem necessary to have it for now.. but I can put back with just the seed storage if you think it'd be useful to still have the seedable abstraction (seems like we only really need the base MobileCoinRng)

Reply:
I think its ok to remove.

I made a suggestion at the end about encapsulating the RNG implementations, the "contextual/seedable" behavior of an RNG could be defined in a protocol, and then RNGs can choose to implement. For this example and chacha20, the context function would return a Data16 representing the word position, this would get set on the PendingTransaction structs returned to consumers, value would be nil for RNGs that don't implement this behavior.

Reply:
> I made a suggestion at the end about encapsulating the RNG implementations

Added some thoughts in a comment after that too that kind of evolve that idea a bit too... marking this resolved, and we can pick up the convo down there.